documentation/src/main/asciidoc/topics/search.adoc

```diff
@@ -0,0 +1,1096 @@
+Query q = qf.create("from com.acme.Book where price > 20");
+
+// Execute the query
+QueryResult<Book> queryResult = q.execute();
+----
+
+
+NOTE: A query will always target a single entity type and is evaluated over the contents of a single cache. Running a
+query over multiple caches or creating queries that target several entity types (joins) is not supported.
+
+Executing the query and fetching the results is as simple as invoking the `run()` method of the _Query_ object. Once
+executed, calling `run()` on the same instance will re-execute the query.
+
+=== Pagination
+
+You can limit the number of returned results by using  the `Query.maxResults(int maxResults)`. This can be used in
+conjunction with `Query.startOffset(long startOffset)` to achieve pagination of the result set.
+
+[source,java]
+----
+// sorted by year and match all books that have "clustering" in their title
+// and return the third page of 10 results
+Query<Book> query = queryFactory.create("FROM com.acme.Book WHERE title like '%clustering%' ORDER BY year").startOffset(20).maxResults(10)
+----
+
+=== Number of Hits
+
+The `QueryResult` object has the `.hitCount()` method to return the total number of results of the query, regardless
+of any pagination parameter. The hit count is only available for indexed queries for performance reasons.
+
+
+=== Iteration
+
+The `Query` object has the `.iterator()` method to obtain the results lazily. It returns an instance of `CloseableIterator` that
+must be closed after usage.
+
+NOTE: Iteration is not currently implemented for Remote Queries
+
+=== Using Named Query Parameters
+
+Instead of building a new Query object for every execution it is possible to include named parameters in the query which
+can be substituted with actual values before execution. This allows a query to be defined once and be efficiently
+executed many times. Parameters can only be used on the right-hand side of an operator and are defined when the query is
+created by supplying an object produced by the _org.infinispan.query.dsl.Expression.param(String paramName)_ method to
+the operator instead of the usual constant value. Once the parameters have been defined they can be set by invoking either
+_Query.setParameter(parameterName, value)_ or _Query.setParameters(parameterMap)_ as shown in the examples below.
+⁠
+[source,java,tile="Using Named Parameters"]
+----
+QueryFactory queryFactory = Search.getQueryFactory(cache);
+// Defining a query to search for various authors and publication years
+Query<Book> query = queryFactory.create("SELECT title FROM com.acme.Book WHERE author = :authorName AND publicationYear = :publicationYear").build();
+
+// Set actual parameter values
+query.setParameter("authorName", "Doe");
+query.setParameter("publicationYear", 2010);
+
+// Execute the query
+List<Book> found = query.list();
+----
+
+Alternatively, multiple parameters may be set at once by supplying a map of actual parameter values:
```

Comment:
It's preferable to address the reader directly and use active voice construction with verb phrases. 

"may be set at once" uses passive voice and obscures the person/thing that performs the action. Also "may" is a modal verb that implies permission more than possibility. It's generally better to prefer "can" or "might" to "may".

Consider revising as follows: "Alternatively, you can supply a map of actual parameter values to set multiple parameters at once"